openvdb_points_houdini/houdini/VRAY_OpenVDB_Points.cc

```diff
@@ -198,6 +201,98 @@ struct GenerateBBoxOp {
+                                const float maxSpeed,
+                                const std::vector<Name>& includeGroups,
+                                const std::vector<Name>& excludeGroups,
+                                const bool collapseVelocityAfter)
+        : mColorIndex(colorIndex)
+        , mVelocityIndex(velocityIndex)
+        , mRamp(ramp)
+        , mMaxSpeed(maxSpeed)
+        , mUseGroups(!includeGroups.empty() || !excludeGroups.empty())
+        , mIncludeGroups(includeGroups)
+        , mExcludeGroups(excludeGroups)
+        , mCollapseVelocityAfter(collapseVelocityAfter) { }
+
+    Vec3f getColorFromRamp(const Vec3T& velocity) const{
+
+        float proportionalSpeed = velocity.length()/(mMaxSpeed);
+        if (proportionalSpeed > 1.0f) proportionalSpeed = 1.0f;
+
+        float rampVal[4];
+        mRamp.rampLookup(proportionalSpeed, rampVal);
+        return Vec3f(rampVal[0], rampVal[1], rampVal[2]);
+    }
+
+    void operator()(LeafRangeT& range) const{
+
+        for (typename LeafRangeT::Iterator leaf=range.begin(); leaf; ++leaf) {
+
+            tools::AttributeWriteHandle<Vec3f>::Ptr colorHandle =
+                tools::AttributeWriteHandle<Vec3f>::create(leaf->attributeArray(mColorIndex));
+
+            typename tools::AttributeWriteHandle<Vec3T>::Ptr velocityHandle =
+                tools::AttributeWriteHandle<Vec3T>::create(leaf->attributeArray(mVelocityIndex));
+
+            const bool uniform = velocityHandle->isUniform();
+            const Vec3f uniformColor = getColorFromRamp(velocityHandle->get(0));
+
+            if (mUseGroups) {
+
+                tools::MultiGroupFilter::Data data(mIncludeGroups, mExcludeGroups);
+                const tools::MultiGroupFilter filter = tools::MultiGroupFilter::create(*leaf, data);
+                tools::FilterIndexIter<IndexOnIter, tools::MultiGroupFilter> filterIndexIter(leaf->beginIndexOn(), filter);
+
+                for (; filterIndexIter; ++filterIndexIter) {
+
+                    Vec3f color = uniform ? uniformColor : getColorFromRamp(velocityHandle->get(Index64(*filterIndexIter)));
+                    colorHandle->set(*filterIndexIter, color);
+                }
+            }
+            else {
+
+                IndexOnIter iter = leaf->beginIndexOn();
+
+                for (; iter; ++iter) {
+
+                    Vec3f color = uniform ? uniformColor : getColorFromRamp(velocityHandle->get(Index64(*iter)));
+                    colorHandle->set(*iter, color);
+                }
+            }
+
+            if (mCollapseVelocityAfter)     velocityHandle->collapse(Vec3T(0));
+        }
+    }
+
+    //////////////////////////////////////////////
+
+    const size_t                            mColorIndex;
+    const size_t                            mVelocityIndex;
+    const UT_Ramp&                          mRamp;
+    const float                             mMaxSpeed;
+    const bool                              mUseGroups;
```

Comment:
It is in the constructor. Now that you point it out though, it wouldn't really cost much to just re-check the contents once per grid (instead of once per op)